pkg/canary/deployment_controller.go

```diff
@@ -129,11 +129,8 @@ func (c *DeploymentController) Promote(cd *flaggerv1.Canary) error {
 			primaryCopy.ObjectMeta.Annotations[k] = v
 		}
 		// update deploy labels
-		primaryCopy.ObjectMeta.Labels = make(map[string]string)
 		filteredLabels := includeLabelsByPrefix(canary.ObjectMeta.Labels, c.includeLabelPrefix)
-		for k, v := range filteredLabels {
-			primaryCopy.ObjectMeta.Labels[k] = v
-		}
+		primaryCopy.ObjectMeta.Labels = makePrimaryLabels(filteredLabels, primaryLabelValue, label)
```

Comment:
could you also make this change in `pkg/canary/daemonset_controller.go`?

Reply:
@aryan9600 sorry I missed that fix, but I have addressed it in the latest